Formula/gnirehtet.rb

```diff
@@ -0,0 +1,67 @@
+class Gnirehtet < Formula
+  desc "Reverse tethering tool for Android"
+  homepage "https://github.com/Genymobile/gnirehtet"
+  url "https://github.com/Genymobile/gnirehtet/archive/v2.3.tar.gz"
+  sha256 "bb2cea28975071b8bdf7bb5dc9536e8119eb0ab113e911cdef7b057bb1ad2582"
+  head "https://github.com/Genymobile/gnirehtet.git"
+
+  depends_on "rust" => :build
+  depends_on "socat" => :test
+
+  resource "java_bundle" do
+    url "https://github.com/Genymobile/gnirehtet/releases/download/v2.3/gnirehtet-java-v2.3.zip"
+    sha256 "93d1d46ee566376596f033832626dd5e89e76c91f2c46d2383735937b7d3b8b0"
+  end
+
+  def install
+    resource("java_bundle").stage { libexec.install "gnirehtet.apk" }
+
+    system "cargo", "install", "--root", libexec, "--path", "relay-rust"
```

Comment:
```suggestion
    system "cargo", "install", "--locked", "--root", libexec, "--path", "relay-rust"
```
Relates to https://github.com/Homebrew/homebrew-core/issues/46025